test/integration/developer_portal/signup_test.rb

```diff
@@ -26,6 +26,44 @@ def test_create
     assert_response :redirect
   end
 
+  def test_create_spam_protected_always
```

Comment:
I think the test are confusing, as they just check that:
- the user is not created
- the response is 200 (I imagine that it's to contrast 301 which is redirection when the user is created successfully)

I'm wondering if it would make more sense to add a corresponding test to https://github.com/3scale/porta/blob/master/features/old/signup/spam_protection.feature

Reply:
There's already a cucumber for that  at: https://github.com/3scale/porta/blob/master/features/old/signup/spam_protection.feature#L27 

I agree this test is not easy to understand, the same for `test_create_spam_protected_suspicious_only`. But how can I check the post request failed?

Reply:
From what I can tell, the tests only verify the cases when spam protection level is set to `captcha` and `none`, but not `auto`(Suspicious only), which is the one that was failing because of the bug and is fixed here.

So, I think the tests for àuto` needs to be added in that feature file.

Reply:
Done!

Reply:
Well done! :+1: 

IMO, I would drop the signup controller tests. It is not clear what exactly they test, and the fix is tested with the new cucumber.
But if you prefer to keep them, maybe it would be nice to refactor a bit to reduce duplication (e.g. the request params which are the same always).

Reply:
I think the tests are needed because they test that a post query can't be sent directly to the controller, bypassing the form, when the spam protection is enabled. The cucumbers doesn't test that, they are testing the form and the overall functionality. About how to check that the spam protection worked, when it works, it returns the form view and the `200 OK`. Asserting that `return code == 200` is confusing, but asserting that the user wasn't created is OK IMO. I added some comments to make it more clear.